qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -211,12 +205,24 @@ def compute_minimum_eigenvalue(
         """
         if not isinstance(self._tmp_ansatz, EvolvedOperatorAnsatz):
             raise QiskitError("The AdaptVQE ansatz must be of the EvolvedOperatorAnsatz type.")
+
         if self._solver.quantum_instance is None:
             raise AlgorithmError(
                 "A QuantumInstance or Backend must be supplied to run the quantum algorithm."
             )
-        # We construct the ansatz once to be able to extract the full set of excitation operators.
-        self._tmp_ansatz._build()
+
+        # construct the expectation
+        if self._solver.expectation is None:
+            expectation = ExpectationFactory.build(
+                operator=operator,
+                backend=self._solver.quantum_instance,
+                include_custom=self._solver.include_custom,
+            )
+        else:
+            expectation = self._solver.expectation
+
+        # Overwrite the solver's ansatz with the initial state
+        self._solver.ansatz = self._tmp_ansatz.initial_state
```

Comment:
This seems a little tricky, what if I plug my VQE instance into AdaptVQE but then decide to use it later on for something else? Would it be possible to just take a copy of the ansatz there? If not, is the ansatz reset to the original value at some point?

Reply:
Well the whole point of the algorithm is to build out the ansatz adaptively. So no, we do not reset it later. In fact, you will see it gets updated again here:
https://github.com/mrossinek/qiskit-terra/blob/9fab0fb262fbec85259c1e6e4c4ce9cf874d4be5/qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py#L265-L267

Reply:
I mean after the whole algorithm is finished 🙂

Reply:
Like when I do
```python
vqe = VQE(my_ansatz, ...)
adapt_vqe = AdaptVQE(vqe, ...)
adapt_sol = adapt_vqe.compute....
vqe_sol = vqe.compute...  # does this still use `my_ansatz`?
```

Reply:
No it will not. It will use the `ansatz` determined by `AdaptVQE`
But maybe this is a change that should be discussed in the main PR and not this PR to a PR...

Reply:
Ok, I have to review it anyways, right? 😛